core/utils/sleeper_task.go

```diff
@@ -83,13 +85,27 @@ func (s *sleeperTask) WakeUp() {
 	}
 }
 
+func (s *sleeperTask) workDone() {
```

Comment:
Is there a way to name these methods differently for the read and write use cases?